Documentation/security/network/encryption-wireguard.rst

```diff
@@ -295,6 +295,45 @@ options:
   - LoadBalancer & NodePort XDP Acceleration
   - Direct Server Return (DSR)
 
+Which traffic encrypted
+=======================
+
+The following table denotes which packets are encrypted with WireGuard depending
+on the mode:
+
++------------------------------------+--------------------+-------------------------+
+| Traffic                            | default mode       | node-to-node encryption |
+|                                    |                    | mode                    |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Pod                  | Y                  | Y                       |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Pod via L7 proxy or  | Y                  | Y                       |
+| L7 ingress service                 |                    |                         |
++------------------------------------+--------------------+-------------------------+
+| Pod to remote Pod via Service with | Y                  | Y                       |
+| kube-proxy-replacement             |                    |                         |
++------------------------------------+--------------------+-------------------------+
+| Pod to Egress Gateway node         | Y                  | Y                       |
```

Comment:
Here we still need https://github.com/cilium/cilium/pull/31586 to be thorough.

The return traffic (gateway node to pod) is also encrypted, unless XDP is enabled.

Reply:
:+1: